### DISCUSSION TITLE
Custom sorting based on YAML frontmatter creation date?

### DISCUSSION THREAD

**Discussion by bloominburges** [SebastianMC/obsidian-custom-sort#179] [Q&A] [answered]
How do i use your add in to sort the files in a folder by the note’s “created” property in my YAML frontmatter? I want it to do it off of this frontmatter property, NOT the file property as the source of truth like some other admins do.

Thanks

**Comment by SebastianMC** [comment]
Hi @bloominburges,

the base idea of sorting specification which you described is:

```yaml
---
sorting-spec: |
  > a-z by-metadata: created
---
```

The above simple sorting spec works under the following assumptions:
- placed in a `sortspec` note in the folder which you want to sort. To sort a folder different than the location of `sortspec` note you'd need to explicitly specify `  target-folder: path-of-target-folder-here`
- the files contain metadata `created` with values which can be ordered alphabetically when treating them as text. That means a date format like yyyy-mm-dd (e.g. 2024-11-16). If the format of values is different, e.g. date in US format mm/dd/yyyy (e.g. 11/16/2024) or - in general - format where day and/or month comes before the year, that obviously won't work. To deal with that type of formatted dates you need a feature requested here #171 (still in development)

**Comment by bloominburges** [comment]
Got it… thanks.
What about if i want reverse? I tried “> z-a by-metadata: created”, and it didn’t work

**Comment by SebastianMC** [comment]
The syntax of sorting order specification in this particular example comprises three elements and two of them are relevant to your question:

1. The **direction of the order** - ascending or descending. In this case the token is `>`
  - `>` stands for descending order, larger values go first. You can use more verbose syntax for it: `order-desc:` as an equivalent for readability
  - `<` can be used for the ascending order, smaller values go first. Verbose equivalent syntax is `order-asc:`

To change the order direction to reverse you need to switch `>` to `<`, and vice versa

2. The **actual sorting order** - in this case the token is `a-z` which represents the _alphabetical_ sorting. It has no reverse equivalent and the direction specification (1) is used to switch between alphabetical and reverse alphabetical.

The spec to use is then:
- `  < a-z by-metadata: created` to put earlier dates first, and
- `  > a-z by-metadata: created` to put later dater first.

**Comment by thomas-bouvier** [comment]
Hi @SebastianMC, thank you for the great plugin!

I am trying to achieve the same thing, except my metadata attribute is named "Last updated at" (containing an ISO date string).

```
---
Created at: 2024-02-24
Last updated at: 2024-12-11
---
```

I used the following `sortspec`:

```
---
sorting-spec: |
  < a-z by-metadata: "Last updated at"
---
```

My notes don't get sorted in the folder. Is it because attribute `Last updated at` contains whitespaces, or am I encountering the same issue as op?

**Comment by SebastianMC** [comment, marked as answer]
Hi @thomas-bouvier,

in the scenario which you described you only need to remove the quotation marks and you should be good:

```yaml
---
sorting-spec: |
  < a-z by-metadata: Last updated at
---
```

The metadata values expressed using ISO date format of YYYY-MM-DD can be used directly for sorting: treating them as plain text for sorting gives correct result.

**Comment by thomas-bouvier** [comment]
I confirm this is working. Thank you. I think this issue can be closed.
